src/com/owncloud/android/ui/activity/ReceiveExternalFilesActivity.java

```diff
@@ -398,11 +404,15 @@ private void populateDirectoryList() {
         if (mFile != null) {
             // TODO Enable when "On Device" is recovered ?
             Vector<OCFile> files = getStorageManager().getFolderContent(mFile/*, false*/);
-            List<HashMap<String, OCFile>> data = new LinkedList<>();
+            files = sortFileList(files);
+
+            List<HashMap<String, Object>> data = new LinkedList<>();
             for (OCFile f : files) {
-                HashMap<String, OCFile> h = new HashMap<>();
+                if (f.isFolder()) {
```

Comment:
Comes from #674, solved by @tobiasKaminsky in #1399.

You reviewed that PR :D :D :D